### ISSUE TITLE
AHT2x exhibiting strange behaviour

### ISSUE THREAD

**Issue by erict-powersoft** [enjoyneering/AHTxx#19]
I've built a few temperature probes that are an ESP8266 with an AHT2x attached. The outdoor AHT20 works fine, indoors I have an AHT20 ('Elderhof') and an AHT25 ('Office') which have weird random bumps. A bump-up in temperature has a corresponding bump-down in the humidity reading.
<img width="486" alt="2023-10-10 08_55_18-Clipboard" src="https://github.com/enjoyneering/AHTxx/assets/13301999/ed587769-ce59-4854-a22a-a838b00a5594">
Readings are taken every 30 seconds and fed into HA over MQTT. I looked at the bump-up of the Office sensor around midnight Oct 10 in detail: It rises over 5 minutes - i.e. it is not instantaneous. It starts to rise at 10:14pm and reaches 2 degrees higher by 10:19pm and the bump ends at 00:43 taking 3 minutes to drop the 2 degrees. There are no heaters or fans going off in that time that could affect the temperature.

Software-wise they're all driven by the same code-base using the arduino architecture.

Any ideas what could be causing this weird behaviour?

Sorry to turn this into a github issue.  Would like to know your opinion as you're most likely an expert with AHT2x devices.
#https://www.reddit.com/r/homeassistant/comments/174ktva/weird_bug_observed_using_aht2x_temperature_sensors/

**Comment by enjoyneering** [comment]
This is all very strange, I have never encountered anything like this. Check what else is turned on in the house at this time. Also place another exactly the same sensor next to the strange one and compare the results.

Use macro AHTXX_USE_READ_DATA on the second (humidity) read. This saves time, because the humidity value is already in the library buffer after the temperature reading. This trick is also reducing sensor self heating.

```
float temp1 = aht10.readTemperature(AHTXX_FORCE_READ_DATA); //AHTXX_FORCE_READ_DATA=read 6-bytes via I2C, takes 80 milliseconds

float humi1 = aht10.readHumidity(AHTXX_USE_READ_DATA); //AHTXX_USE_READ_DATA=use 6-bytes from humidity reading, takes zero milliseconds!!!
```

**Comment by erict-powersoft** [comment]
Hi, thanks for your reply.  I have since realized that I was doing 2 successive reads, one for Hum, and one for Temp.  I've now reduced to a single read with AHTXX_USE_READ_DATA, and I've also lengthened the in-between pause from 30 seconds to 1 minute. This has made a huge difference, but there are still occasional short spikes in temperature with corresponding drops in humidity.  See new image here: [https://www.reddit.com/media?url=https%3A%2F%2Fpreview.redd.it%2Fvsa4x6a9ahub1.png%3Fwidth%3D1320%26format%3Dpng%26auto%3Dwebp%26s%3D1f851f21343ecf15dbbeb1179f5355498f12ea04](url)
The fact that there's an identical temperature spike and humidty trough at the same time means something is still weird.
  There's nothing else in the house other than 2 cats (we're currently travelling) and the daily house-sitter human -- none have ever been observed playing with the sensors.  
PS: When I changed the code in the "Elderhof" sensor, I must have killed the AHT20, as it's now dead.  So there's only a single device that exhibits this behaviour...
